test/test_linalg.py

```diff
@@ -922,6 +923,90 @@ def test_nuclear_norm_exceptions_old(self, device):
         self.assertRaisesRegex(IndexError, "Dimension out of range", torch.norm, x, "nuc", (0, 2))
 
 
+    @precisionOverride({torch.float: 1e-2, torch.cfloat: 1e-4})
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypesIfCPU(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @dtypesIfCUDA(torch.float32, torch.float64)
+    def test_cholesky(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+
+        def run_test(shape, batch):
+            A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+            expected_L = np.linalg.cholesky(A.cpu().numpy())
+            actual_L = torch.linalg.cholesky(A)
+            self.assertEqual(actual_L, expected_L)
+
+        shapes = (0, 3, 5)
+        batches = ((), (3, ), (2, 2))
+        larger_input_case = [(100, (5, ))]
+        for shape, batch in list(itertools.product(shapes, batches)) + larger_input_case:
+            run_test(shape, batch)
```

Comment:
Would you add the complex dtypes for CUDA and assert that an error is thrown. Something like: 

```
if self.device_type == 'cuda' and dtype.is_complex:
  with self.assertRaises(RuntimeError):
    run_test(shape, batch)
else:
   run_test(shape, batch)
```

Reply:
Nevermind. I see you have a separate test for this below. That's totally fine.